tests/bundle/Integration/BaseTest.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\IbexaScheduledVisibility\Tests\Integration;
+
+use DateTime;
+use Ibexa\Contracts\Core\Repository\Values\Content\Content;
+use Ibexa\Contracts\Core\Repository\Values\ContentType\ContentType;
+use Ibexa\Tests\Integration\Core\Repository\BaseTest as APIBaseTest;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Service\ScheduledVisibilityService;
+
+abstract class BaseTest extends APIBaseTest
+{
+    protected function provideCases(): iterable
+    {
+        return [
+            [
+                [
+                    'publish_from' => null,
+                    'publish_to' => null,
+                ],
+                false,
+            ],
+            [
+                [
+                    'publish_from' => null,
+                    'publish_to' => new DateTime('tomorrow'),
+                ],
+                false,
+            ],
+            [
+                [
+                    'publish_from' => null,
+                    'publish_to' => new DateTime('yesterday'),
+                ],
+                true,
+            ],
+            [
+                [
+                    'publish_from' => new DateTime('2 days ago'),
+                    'publish_to' => new DateTime('yesterday'),
+                ],
+                true,
+            ],
+            [
+                [
+                    'publish_from' => new DateTime('yesterday'),
+                    'publish_to' => new DateTime('tomorrow'),
+                ],
+                false,
+            ],
+            [
+                [
+                    'publish_from' => new DateTime('tomorrow'),
+                    'publish_to' => null,
+                ],
+                true,
+            ],
+            [
+                [
+                    'publish_from' => new DateTime('tomorrow'),
+                    'publish_to' => new DateTime('2 day'),
+                ],
+                true,
+            ],
+        ];
+    }
+
+    protected function getScheduledVisibilityService(): ScheduledVisibilityService
+    {
+        /** @var ScheduledVisibilityService $service */
+        $service = $this->getSetupFactory()->getServiceContainer()->get('Netgen\Bundle\IbexaScheduledVisibilityBundle\Service\ScheduledVisibilityService');
```

Comment:
We should use `::class` constants.

Reply:
fixed in 90af924cc2dfb3ca797b8877778bb3b95d98291f